core/storage/merkle/polkadot_trie_db/polkadot_trie_db.hpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_STORAGE_MERKLE_POLKADOT_TRIE_DB_POLKADOT_TRIE_DB_HPP_
+#define KAGOME_CORE_STORAGE_MERKLE_POLKADOT_TRIE_DB_POLKADOT_TRIE_DB_HPP_
+
+#include <map>
+#include <memory>
+#include <optional>
+
+#include "crypto/hasher.hpp"
+#include "storage/merkle/polkadot_trie_db/polkadot_codec.hpp"
+#include "storage/merkle/polkadot_trie_db/polkadot_node.hpp"
+#include "storage/merkle/trie_db.hpp"
+
+namespace kagome::storage::merkle {
+
+  class PolkadotTrieDb : public TrieDb {
+    using MapCursor = face::MapCursor<common::Buffer, common::Buffer>;
+    using WriteBatch = face::WriteBatch<common::Buffer, common::Buffer>;
+    using Codec = scale::ScaleCodec<common::Buffer>;
+    using NodePtr = std::shared_ptr<PolkadotNode>;
+    using BranchPtr = std::shared_ptr<BranchNode>;
+
+    // to output the trie into a stream
+    template <typename Stream>
+    friend Stream &operator<<(Stream &s, const PolkadotTrieDb &trie);
+    template <typename Stream>
+    friend Stream &printNode(Stream &s, NodePtr node,
+                             const PolkadotTrieDb &trie, size_t nest_level = 0);
```

Comment:
nest_level? 
next_level?

Reply:
Nope, it's for nested nodes, to display hierarchy in the trie.